AdobeIms/Test/Unit/Model/ConfigTest.php

```diff
@@ -36,28 +35,129 @@ class ConfigTest extends TestCase
     private $scopeConfigMock;
 
     /**
-     * Prepare test objects.
+     * @var UrlInterface|MockObject
+     */
+    private $urlMock;
+
+    /**
+     * Set up test mock objects
      */
     protected function setUp(): void
     {
-        $this->objectManager = new ObjectManager($this);
         $this->scopeConfigMock = $this->createMock(ScopeConfigInterface::class);
-        $this->config = $this->objectManager->getObject(
-            Config::class,
-            [
-                'scopeConfig' => $this->scopeConfigMock
-            ]
-        );
+        $this->urlMock = $this->createMock(UrlInterface::class);
+
+        $this->config = new Config($this->scopeConfigMock, $this->urlMock);
     }
 
     /**
-     * Get API key test.
+     * Test for \Magento\AdobeIms\Model\Config::getApiKey
      */
     public function testGetApiKey(): void
     {
         $this->scopeConfigMock->expects($this->once())
             ->method('getValue')
-            ->with(self::CONFIG_XML_PATH_API_KEY);
-        $this->config->getApiKey();
+            ->with(Config::XML_PATH_API_KEY)
+            ->willReturn(self::STRING_RETURN);
+
+        $this->assertEquals(self::STRING_RETURN, $this->config->getApiKey());
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getPrivateKey
+     */
+    public function testGetPrivateKey(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
```

Comment:
I think it's better not to check for the number of calls, just check the output.
```suggestion
        $this->scopeConfigMock->method('getValue')
             ->with(Config::XML_PATH_PRIVATE_KEY)
             ->willReturn(self::STRING_RETURN);
```
and in all other tests as well

Reply:
Sure, I will do that. But why do we not want to check for number of calls as well? It further makes the test solid, right? Or are there any specific situations where call frequency is not recommended?

Reply:
Fixed this, thanks!